src/oauth2c.erl

```diff
@@ -146,7 +104,7 @@ retrieve_access_token(Type, Url, ID, Secret, Scope, Options) ->
              ,secret    = Secret
              ,scope     = Scope
              },
-  do_retrieve_access_token(Client, Options).
+  get_token(Client, Options, whereis(?TOKEN_CACHE_SERVER)).
```

Comment:
I think the user of oauth2_client should choose if they want to cache tokens or not.

Also the cache gen_server is now part of the application and will always be running as long as the application is started. So I would instead hook the logic regarding if the token should be cached or not to an option passed on the Options parameter.

Reply:
Sounds like a good idea, what should the default? On or Off?

Reply:
Default should be off since that's the current behaviour.

Reply:
Okay, I will fix it!

Reply:
Options is currently a list, would it not make more sense if it would be a Proplist?

Reply:
I think the same type should be used in both.

Reply:
After making some changes, it is now required to invoke retrieve_access_token like below to enable caching:

```
       oauth2c:retrieve_access_token(?CLIENT_CREDENTIALS_GRANT,
                                                  ?AUTH_URL,
                                                  <<"ID">>,
                                                  <<"SECRET">>,
                                                  undefined,
                                                  [{enable_cache, true}])
```

and it is also possible to pass the enable_cache property to the request function as well. 

At the moment if enable_cache is not set, then oauth2c will neither insert or lookup values in the cache. It will directly call do_retreive_access token like the previous implementation. Is this the behaviour we want or do we want the to always look in the cache for a token but only insert values if the enable_cache property is true (but then we would change the property name to something like "cache_token") ?

Reply:
hmmm that's actually a good question. What would cause the least surprise to the developer? I guess in this case it doesn't actually matter since we are hiding the retrieval of the token from the user. So I say for you to choose the one that makes the code simpler/easier to read/smaller.

Reply:
I think I will go with complete disabling of the cache by default since it will not be evident to users of the library that caching is actually taking place. This approach also leads to simpler and cleaner code.